src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -759,5 +763,90 @@ private static bool IsPotentialFilterCharacter(char c)
                 || char.IsNumber(c)
                 || c == '_';
         }
+
+        private static (int index, bool forceHardSelection) GetDefaultsMatch(
+            string filterText,
+            ImmutableArray<(RoslynCompletionItem item, PatternMatch? patterMatch)> itemsWithMatch,
+            int selectedIndex,
+            ImmutableArray<string> defaults,
+            bool aggressive)
+        {
+            // We only preselect when we are very confident with the selection, so don't override it.
+            if (defaults.IsDefaultOrEmpty || itemsWithMatch[selectedIndex].item.Rules.MatchPriority >= MatchPriority.Preselect)
+                return (selectedIndex, false);
+
+            if (aggressive)
```

Comment:
where does the term aggressive come from?

Reply:
we need comments exaplining what 'aggressive' means and what the intuition is here as to how this should behave.